components/bt/host/bluedroid/api/include/api/esp_gap_ble_api.h

```diff
@@ -17,2687 +17,3971 @@
+
+/* if Bluetooth Low Energy 5.0 Feature Support */  
     #if (BLE_50_FEATURE_SUPPORT == TRUE)
-    BLE_SCAN_DUPLICATE_ENABLE_RESET,             /*!< Duplicate filtering enabled, reset for each scan period, only supported in BLE 5.0. */
-    #endif
+    BLE_SCAN_DUPLICATE_ENABLE_RESET,             /*!< Duplicate filtering enabled, reset for each scan period, only supported in bluetooth low energy 5.0. */
+    #endif// #if (BLE_50_FEATURE_SUPPORT == TRUE)
     BLE_SCAN_DUPLICATE_MAX                       /*!< Reserved for future use. */
 } esp_ble_scan_duplicate_t;
+
+// if Bluetooth Low Energy 4.2 Feature Support
 #if (BLE_42_FEATURE_SUPPORT == TRUE)
-/// Ble scan parameters
+
+/**
+ * @brief Bluetooth Low Energy (LE) scan parameters
+ * - See more in BLUETOOTH CORE SPECIFICATION Version 5.4 Vol 6, Part B.
+ */
 typedef struct {
-    esp_ble_scan_type_t     scan_type;              /*!< Scan type */
-    esp_ble_addr_type_t     own_addr_type;          /*!< Owner address type */
-    esp_ble_scan_filter_t   scan_filter_policy;     /*!< Scan filter policy */
-    uint16_t                scan_interval;          /*!< Scan interval. This is defined as the time interval from
-                                                      when the Controller started its last LE scan until it begins the subsequent LE scan.
-                                                      Range: 0x0004 to 0x4000 Default: 0x0010 (10 ms)
-                                                      Time = N * 0.625 msec
-                                                      Time Range: 2.5 msec to 10.24 seconds*/
-    uint16_t                scan_window;            /*!< Scan window. The duration of the LE scan. LE_Scan_Window
-                                                      shall be less than or equal to LE_Scan_Interval
-                                                      Range: 0x0004 to 0x4000 Default: 0x0010 (10 ms)
-                                                      Time = N * 0.625 msec
-                                                      Time Range: 2.5 msec to 10240 msec */
-    esp_ble_scan_duplicate_t  scan_duplicate;       /*!< The Scan_Duplicates parameter controls whether the Link Layer should filter out
-                                                        duplicate advertising reports (BLE_SCAN_DUPLICATE_ENABLE) to the Host, or if the Link Layer should generate
-                                                        advertising reports for each packet received */
+    esp_ble_scan_type_t      scan_type;              /*!< Scan type, specifies whether to use passive or active. */
+    esp_ble_addr_type_t      own_addr_type;          /*!< Owner address type, specifies the type of Bluetooth address used by the scanning device. */
+    esp_ble_scan_filter_t    scan_filter_policy;     /*!< Scan filter policy, the scanning filter policy shall apply when receiving an advertising Protocol Data Unit (PDU) or scan response PDU when scanning. */
```

Comment:
the scanning filter policy will be applied when receiving advertising packet or scan response packet.